pkg/utils/tokenizer/interfaces.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2025 The Aibrix Team.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tokenizer
+
+import "context"
+
+// Tokenizer is the basic tokenizer interface for backward compatibility
+type Tokenizer interface {
+	TokenizeInputText(string) ([]byte, error)
+}
+
+// extendedTokenizer represents an extended tokenizer interface with advanced features
```

Comment:
Extend the comment for some examples of `advanced features`. Also I'm not sure if you need an `extendedTokenizer` between `Tokenizer` and `remoteTokenizer`. Why not subclass `remoteTokenizer` directly from `Tokenizer`?